src/serialization/UntypedNode.cs

```diff
@@ -0,0 +1,41 @@
+﻿// ------------------------------------------------------------------------------
+//  Copyright (c) Microsoft Corporation.  All Rights Reserved.  Licensed under the MIT License.  See License in the project root for license information.
+// ------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+
+namespace Microsoft.Kiota.Abstractions.Serialization
+{
+    /// <summary>
+    /// Base class for untyped node.
+    /// </summary>
+    public class UntypedNode : IParsable
```

Comment:
I believe the goal was to make this one generic and abstract? or did I miss something @andrueastman?

Reply:
yes, my concern with having everything return object is that we, and the SDK users, will have to go through additional casting.

Reply:
Totally agree on that. I was suggesting moving to a method which will resolve this and provide a consistent way across the langs using covariant return values. See comment at https://github.com/microsoft/kiota-abstractions-dotnet/pull/177/files#r1472874629